homeassistant/components/xiaomi_miio/binary_sensor.py

```diff
@@ -151,17 +163,30 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
         for description in BINARY_SENSOR_TYPES:
             if description.key not in sensors:
                 continue
-            entities.append(
-                XiaomiGenericBinarySensor(
-                    f"{config_entry.title} {description.name}",
-                    hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE],
-                    config_entry,
-                    f"{description.key}_{config_entry.unique_id}",
-                    hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR],
-                    description,
-                )
+            sensor = XiaomiGenericBinarySensor(
```

Comment:
It's nice if we can avoid creating the entity if the data value is None and we won't add the entity.